src/cookies/analytics.ts

```diff
@@ -1,36 +1,34 @@
+import { getCookieValue } from './utils/cookie';
+
 export const ANALYTICS_COOKIE_NAME = 'stardogAnalyticsConsent';
 
 export type AnalyticsConsent = {
   consented: boolean;
   identity?: string;
 };
 
-export const getCookies = (): string => {
-  return document.cookie;
-};
-
-export const getCookieValue = (cookieName: string): string => {
-  const cookies = getCookies();
-  const match = cookies.match(`(^|;)\\s*${cookieName}\\s*=\\s*([^;]+)`);
-  return match ? (match.pop() as string) : '';
-};
-
 export const decodeAnalyticsCookie = (rawCookie: string): AnalyticsConsent => {
   const defaultConsent: AnalyticsConsent = {
     consented: false,
   };
   try {
     const decoded = JSON.parse(atob(rawCookie));
+
     const consented: boolean = decoded.consented;
-    const identity: string = decoded.identity || '';
-    if (typeof identity !== 'string' || typeof consented !== 'boolean') {
+    const identity: string = decoded.identity;
+
+    if (
+      typeof consented !== 'boolean' ||
+      (typeof identity !== 'undefined' && typeof identity !== 'string')
```

Comment:
Maybe we can make this cleaner with a check if it is set first like:

```
const optionalTypeCheck(val: any, type: string): bool => {
   return typeof val !== 'undefined' && typeof val !== type;
}
```

Reply:
how's this
[`0e8ef5c` (#7)](https://github.com/stardog-union/portal-sdk/pull/7/commits/0e8ef5cd4a14e858b40c884d1442065a99a430ca)